lib/facter/rhsm.rb

```diff
@@ -31,6 +32,11 @@
     end
   end
 
+  # Find any syspurpose defined
+  chunk(:syspurpose) do
+    Puppet::Util::Json.load_file_if_valid('/etc/rhsm/syspurpose/syspurpose.json') if File.exist? '/etc/rhsm/syspurpose/syspurpose.json'
```

Comment:
I checked some servers and they all have the file with content `{}`. do you want to use the file in that case and basically create an empty fact?

Reply:
I'm not sure what feels more natural for querying the settings....

On the one hand, wraping everything in `if X in Y` twice (if file has content, if setting is set) is a bit ugly.  On the other, a fact whose value is `{}` is really meaningless data.

Reply:
After thinking on it a bit, folks storing the data in puppetdb would probably not have the fact set to `{}`